### PR TITLE
Handle arrays

### PR DESCRIPTION
**Pull request by brauliobo**
Hi @arturoherrero,

Please review this PR. It has the handling of arrays we talked about.

Some code style things are different.
Let me know of any needed changes

### CODE REVIEW

File: lib/ofstruct.rb [hunk 1]
```diff
@@ -50,11 +50,18 @@ def method_missing(name, *args)
...
 
-  def assign(key, value)
-    if value.is_a?(Hash)
-      @members[key.to_sym] = self.class.new(value)
+  def convert_value(value)
+    case value
+    when Hash
+      self.class.new(value)
+    when Array
+      value.map{ |v| convert_value v }
```
Comment (arturoherrero) [inline lib/ofstruct.rb]:
```suggestion
      value.map { |element| convert_value(element) }
```

File: lib/ofstruct.rb [hunk 2]
```diff
@@ -50,11 +50,18 @@ def method_missing(name, *args)
...
     else
-      @members[key.to_sym] = value
+      value
     end
   end
+
+  def assign(key, value)
+    @members[key.to_sym] = convert_value value
```
Comment (arturoherrero) [inline lib/ofstruct.rb]:
```suggestion
    @members[key.to_sym] = convert_value(value)
```

File: spec/lib/ofstruct_spec.rb [hunk 3]
```diff
@@ -43,6 +43,7 @@
       let(:symbol_args) { { :name => "John" } }
       let(:string_args) { { "name" => "John" } }
       let(:nested_args) { { person: { name: "John" } } }
+      let(:array_args)  { { list: [{ name: "John" }, { name: 'Doe' } ] } }
```
Comment (arturoherrero) [inline spec/lib/ofstruct_spec.rb]:
```suggestion
      let(:array_args)  { { list: [{ name: "John" }, { name: 'Doe' }] } }
```

File: spec/lib/ofstruct_spec.rb [hunk 4]
```diff
@@ -52,6 +53,14 @@
         end
       end
 
+      context "with a array of hashes" do
+        let(:args) { array_args }
+
+        it "works with reader" do
+          expect(ofstruct.list.map(&:name)).to eq(['John', 'Doe'])
```
Comment (arturoherrero) [inline spec/lib/ofstruct_spec.rb]:
```suggestion
          expect(ofstruct.list.map(&:name)).to include('John', 'Doe')
```

**Review by arturoherrero — COMMENTED** [review]
Thanks for creating this. I left a few stylish suggestions, but overall, it looks good.

There are other things that we can refactor for consistency and improve the specs: covering _black hole_ objects, `works with reader` is not very clear, etc. but all of this is outside the scope of this merge request.

I will improve a little bit more the code after this get merged.

**Comment by brauliobo** [comment]
just applied all suggestions

**Comment by arturoherrero** [comment]
Let's merge this one, and I will refactor a little bit the code before moving to the other issues.

Thanks for your contribution, @brauliobo!

**Pull request merged**